### ISSUE TITLE
Implementing Adversarial Latent Autoencoders [CVPR2020]

### ISSUE THREAD

**Issue by frostbyte012** [keras-team/keras-cv#404]
**Short Description**

Autoencoder networks are unsupervised approaches aiming at combining generative and representational properties by learning simultaneously an encoder-generator map. 

**Papers**

https://arxiv.org/pdf/2004.04467.pdf

**Existing Implementations**

https://github.com/podgorskiy/ALAE

**Other Information**

One implementation will ensure creating 1024x1024
<br>
The resolution can also produce face reconstructions and manipulations based on real images.
<img src="https://camo.githubusercontent.com/0051543675c22f1ec956c0c443b163ceb7804aa819fd98c8550118a1bed3b13e/68747470733a2f2f706f64676f72736b69792e636f6d2f7374617469632f7265636f6e737472756374696f6e735f6d756c74697265736f6c7574696f6e5f322e6a7067" alt="Your image url" width="1000" height="250">
<br>
<img src="https://user-images.githubusercontent.com/3229783/79670218-63080d80-818f-11ea-9e50-927b8af3e7b5.gif" alt="Your image url" width="400" height="200"><img src="https://user-images.githubusercontent.com/3229783/79530431-4bb90b00-803d-11ea-9ce3-25dfc3df253a.gif" alt="Your image url" width="400" height="200">
<br>
<img src="https://camo.githubusercontent.com/4e597609c16803e68e88d1097f4394bea16a51e590143631d51bb079da101dcd/68747470733a2f2f706f64676f72736b69792e636f6d2f7374617469632f7374796c656d69782e6a7067" alt="Your image url" width="1000" height="700">

**Comment by frostbyte012** [comment]
@innat, @LukeWood was looking by some awesome examples and came by it, can you all confirm the feasibility of this implementation to be added here in Keras CV ?

**Comment by LukeWood** [comment]
I don't think that we have anything on the roadmap for offering algorithms like this.  I'd guess we are more likely to offer SimCLR or something like that than this in the near future.

**Comment by innat** [comment]
@frostbyte012 
It might possible to add if keras-cv start supporting GAN based model in the future, [LIKE](https://github.com/eriklindernoren/Keras-GAN). 

@LukeWood 
I think [TF-Similarity](https://github.com/tensorflow/similarity#supported-algorithms) already offers SimCLR and mutual algorithms.

**Comment by frostbyte012** [comment]
@LukeWood Sure Sir! But I'd love to see it implement in kerasCV soon someday!
